src/main/java/uk/ac/ebi/ampt2d/accession/serial/block/MonotonicRange.java

```diff
@@ -0,0 +1,161 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.serial.block;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MonotonicRange implements Comparable<MonotonicRange> {
+
+    private final long start;
+
+    private final long end;
+
+    public MonotonicRange(long start, long end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    public int compareTo(MonotonicRange monotonicRange) {
+        return Long.compare(start, monotonicRange.start);
+    }
+
+    /**
+     * Returns a array of monotonically increasing ids from start to end inclusively. The array is generated and
+     * returned on demand to make the life span of the array as short as possible in order to have a smaller memory
+     * footprint
+     *
+     * @return
+     */
+    public long[] getIds() {
+        int size = getTotalOfValues();
+        long[] ids = new long[size];
+        long tempId = start;
+        for (int i = 0; i < size; i++) {
+            ids[i] = tempId;
+            tempId++;
+        }
+        return ids;
+    }
+
+    public long getStart() {
+        return start;
+    }
+
+    public long getEnd() {
+        return end;
+    }
+
+    public int getTotalOfValues() {
+        return Math.toIntExact((end - start + 1));
+    }
+
+    public MonotonicRange splitLeft(int numberOfElements) {
+        return new MonotonicRange(start, start + numberOfElements - 1);
+    }
+
+    public MonotonicRange splitRight(int numberOfElements) {
+        return new MonotonicRange(start + numberOfElements, end);
+    }
+
+    public static List<MonotonicRange> convertToMonotonicRanges(long... accessions) {
+        List<MonotonicRange> ranges = new ArrayList<>();
+
+        if (accessions.length == 0) {
+            return ranges;
+        }
+
+        Arrays.sort(accessions);
+        long minMonotonicValue = accessions[0];
+        long maxMonotonicValue = accessions[0];
+        for (int i = 1; i < accessions.length; i++) {
+            if (maxMonotonicValue + 1 == accessions[i]) {
+                maxMonotonicValue = accessions[i];
+            } else {
+                ranges.add(new MonotonicRange(minMonotonicValue, maxMonotonicValue));
+                minMonotonicValue = accessions[i];
+                maxMonotonicValue = accessions[i];
+            }
+        }
+        ranges.add(new MonotonicRange(minMonotonicValue, maxMonotonicValue));
+        return ranges;
+    }
+
+    public List<MonotonicRange> exclude(List<MonotonicRange> ranges) {
```

Comment:
I don't understand what this method is used for. I've simulated it using several intersecting and not intersecting ranges and it returns an empty list.

Reply:
At first I didn't understand what the method was for either. Maybe we could rename it to something like `excludeIntersections`?